File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3040_maximum-number-of-operations-with-the-same-score-ii/solutions/python3/solution.py
from collections import Counter
from typing import List

class Solution:
    def maxOperations(self, nums: List[int]) -> int:
        # Counting all possible scores
        n = len(nums)
        score_count = Counter()
        for i in range(n // 2):
            score_front = nums[i] + nums[i + 1]  # First two elements from start
            score_back = nums[n - 1 - i] + nums[n - 2 - i]  # Last two elements from end
            score_cross = nums[i] + nums[n - 1 - i]  # First and last elements
            score_count[score_front] += 1
            if i != n - 2 - i:
                score_count[score_back] += 1
            score_count[score_cross] += 1

        # Finding the maximum number of operations with the same score
        max_operations = 0
        for score, count in score_count.items():
            max_operations = max(max_operations, count)

        return max_operations

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3040_maximum-number-of-operations-with-the-same-score-ii/solutions/c/solution.c
#include <stdlib.h>

int maxOperations(int* nums, int numsSize) {
    int* pairs = (int*)calloc(2001, sizeof(int)); // Maximum possible sum is 2000 (1000 + 1000)
    int maxOps = 0;

    // Fill in the count for all possible pair sums
    for (int i = 0; i < numsSize; i++) {
        for (int j = i + 1; j < numsSize; j++) {
            int sum = nums[i] + nums[j];
            pairs[sum]++;
        }
    }

    // Check the maximum number of same-score operations we can perform
    for (int sum = 2; sum <= 2000; sum++) {
        int operations = pairs[sum];
        if (operations > maxOps) {
            maxOps = operations;
        }
    }

    free(pairs);
    return maxOps;
}
